[0.11.0-pre-548] Unable to change mod configs through Mod Menu

Using Fabric.
![image](https://user-images.githubusercontent.com/37044997/215592632-44a15def-f636-4564-ad3f-57399bb6380f.png)
(no config button in the top right)

Full traceback:
```
[18:05:42] [Render thread/ERROR]: Error from mod 'hexcasting'
java.lang.RuntimeException: java.lang.NoSuchMethodException: net.minecraft.class_2960.<init>()
	at me.shedaniel.autoconfig.util.Utils.constructUnsafely(Utils.java:49) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.DefaultGuiProviders.lambda$apply$44(DefaultGuiProviders.java:242) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.registry.GuiRegistry.lambda$get$3(GuiRegistry.java:74) ~[cloth-config-11.1.118-fabric.jar:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at me.shedaniel.autoconfig.gui.registry.GuiRegistry.get(GuiRegistry.java:74) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.registry.ComposedGuiRegistryAccess.lambda$get$0(ComposedGuiRegistryAccess.java:49) ~[cloth-config-11.1.118-fabric.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1002) ~[?:?]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647) ~[?:?]
	at me.shedaniel.autoconfig.gui.registry.ComposedGuiRegistryAccess.get(ComposedGuiRegistryAccess.java:51) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.registry.api.GuiRegistryAccess.getAndTransform(GuiRegistryAccess.java:38) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.DefaultGuiProviders.lambda$getChildren$96(DefaultGuiProviders.java:538) ~[cloth-config-11.1.118-fabric.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
	at me.shedaniel.autoconfig.gui.DefaultGuiProviders.getChildren(DefaultGuiProviders.java:543) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.DefaultGuiProviders.getChildren(DefaultGuiProviders.java:530) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.registry.GuiRegistry.lambda$get$3(GuiRegistry.java:74) ~[cloth-config-11.1.118-fabric.jar:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at me.shedaniel.autoconfig.gui.registry.GuiRegistry.get(GuiRegistry.java:74) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.registry.ComposedGuiRegistryAccess.lambda$get$0(ComposedGuiRegistryAccess.java:49) ~[cloth-config-11.1.118-fabric.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1002) ~[?:?]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647) ~[?:?]
	at me.shedaniel.autoconfig.gui.registry.ComposedGuiRegistryAccess.get(ComposedGuiRegistryAccess.java:51) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.registry.api.GuiRegistryAccess.getAndTransform(GuiRegistryAccess.java:38) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.ConfigScreenProvider.lambda$get$5(ConfigScreenProvider.java:129) ~[cloth-config-11.1.118-fabric.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at me.shedaniel.autoconfig.gui.ConfigScreenProvider.lambda$get$6(ConfigScreenProvider.java:126) ~[cloth-config-11.1.118-fabric.jar:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[?:?]
	at me.shedaniel.autoconfig.gui.ConfigScreenProvider.get(ConfigScreenProvider.java:125) ~[cloth-config-11.1.118-fabric.jar:?]
	at me.shedaniel.autoconfig.gui.ConfigScreenProvider.get(ConfigScreenProvider.java:45) ~[cloth-config-11.1.118-fabric.jar:?]
	at at.petrak.hexcasting.fabric.interop.ModMenuInterop.lambda$getModConfigScreenFactory$0(ModMenuInterop.java:14) ~[hexcasting-fabric-1.20.1-0.11.1-7.jar:?]
	at com.terraformersmc.modmenu.ModMenu.getConfigScreen(ModMenu.java:61) ~[modmenu-7.2.2.jar:?]
	at com.terraformersmc.modmenu.gui.ModsScreen.method_25426(ModsScreen.java:114) ~[modmenu-7.2.2.jar:?]
	at net.minecraft.class_437.method_25423(class_437.java:297) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1507(class_310.java:1080) ~[client-intermediary.jar:?]
	at com.terraformersmc.modmenu.gui.widget.ModMenuButtonWidget.lambda$new$0(ModMenuButtonWidget.java:15) ~[modmenu-7.2.2.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:56) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:189) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[client-intermediary.jar:?]
	at net.minecraft.class_442.method_25402(class_442.java:299) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:409) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:169) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:102) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:169) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:?]
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3474) ~[lwjgl-glfw-3.3.1.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:237) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1244) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:802) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[minecraft-1.20.1-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.11.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[NewLaunch.jar:?]
Caused by: java.lang.NoSuchMethodException: net.minecraft.class_2960.<init>()
	at java.lang.Class.getConstructor0(Class.java:3617) ~[?:?]
	at java.lang.Class.getDeclaredConstructor(Class.java:2786) ~[?:?]
	at me.shedaniel.autoconfig.util.Utils.constructUnsafely(Utils.java:45) ~[cloth-config-11.1.118-fabric.jar:?]
	... 77 more
```

--- a/Fabric/src/main/java/at/petrak/hexcasting/fabric/FabricHexConfig.java
+++ b/Fabric/src/main/java/at/petrak/hexcasting/fabric/FabricHexConfig.java
@@ -18,6 +18,7 @@ import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.Mth;
 import net.minecraft.world.level.Level;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import static at.petrak.hexcasting.api.mod.HexConfig.anyMatchResLoc;
@@ -199,7 +200,12 @@ public class FabricHexConfig extends PartitioningSerializer.GlobalData {
 
         // TODO: hook this up to the config, change Jankery, test, also test scroll injects on fabric
         @ConfigEntry.Gui.Tooltip
-        private List<ResourceLocation> loreInjections = HexLootHandler.DEFAULT_LORE_INJECTS;
+        private List<String> loreInjectionsRaw = HexLootHandler.DEFAULT_LORE_INJECTS
+                .stream()
+                .map(ResourceLocation::toString)
+                .toList();
+        @ConfigEntry.Gui.Excluded
+        private transient List<ResourceLocation> loreInjections;
         @ConfigEntry.Gui.Tooltip
         private double loreChance = HexLootHandler.DEFAULT_LORE_CHANCE;
 
@@ -220,6 +226,16 @@ public class FabricHexConfig extends PartitioningSerializer.GlobalData {
 
             } catch (Exception e) {
                 throw new ValidationException("Bad parsing of scroll injects", e);
+            }
+
+            this.loreInjections = new ArrayList<>();
+            try {
+                for (var table : this.loreInjectionsRaw) {
+                    ResourceLocation loc = new ResourceLocation(table);
+                    this.loreInjections.add(loc);
+                }
+            } catch (Exception e) {
+                throw new ValidationException("Bad parsing of lore injects", e);
             }
 
             this.loreChance = Mth.clamp(this.loreChance, 0.0, 1.0);

